app/src/utils/hooks.ts

```diff
@@ -105,14 +106,36 @@ export const useElementDimensions = (): [RefObject<HTMLElement>, Dimensions | un
 };
 
 export const usePageParams = () => {
-  const [pageParams, setPageParams] = useQueryParams({
-    page: NumberParam,
-    pageSize: NumberParam,
-  });
+  const router = useRouter();
```

Comment:
Just out of curiousity, what is this fixing?

Reply:
Previously we were overwriting all query params (meaning setting page number would remove the active tab param). When I stopped overwriting all params, our code then added `id` (for fine tune id) to the query params, which was long and annoying. NextJS's built-in router handles both these problems well.